### ISSUE TITLE
FileSink constructor checking only path for null

### ISSUE THREAD

**Issue by reubenalfred** [serilog/serilog-sinks-file#287]
Hi,

I was just browsing the FileSink file and noticed that the path parameter is only tested for null and doesn't include a test for NullOrWhiteSpace.

Further down in the constructor, it is tested to see if the path's dir is null or white space and creates the directory if the path string is valid, however, no action is specified when the string is empty.

it is an internal constructor, perhaps it may not be that important. i thought i'd let you know...

regards, reuben

![filesink](https://github.com/serilog/serilog-sinks-file/assets/115092608/0586fcc5-65b1-42eb-b378-dc49d23eed7e)

**Comment by reubenalfred** [comment]
if it needs fixing ill be happy to do that :)

**Comment by nblumhardt** [comment]
Hi @reubenalfred! `string.IsNullOrWhiteSpace("")` is `true` so I don't think the empty string case is neglected here.

**Comment by reubenalfred** [comment]
If it is null or white space it won't create the directory sure but it will still attempt to execute the rest of the code without a definite directory.

**Comment by reubenalfred** [comment]
> > action is specified when the string is empty
> 
> `string.IsNullOrWhiteSpace`: "true if the value parameter is null or **Empty**, or if value consists exclusively of white-space characters." [doc](https://learn.microsoft.com/en-us/dotnet/api/system.string.isnullorwhitespace?view=net-7.0)

Where is the action specified? Please show me??

**Comment by reubenalfred** [comment]
Cool will do that, I personally like to test for obvious and explicitly throw exceptions rather than rely on the API's otherwise it is too much to remember what throws what!

**Comment by bartelink** [comment]
@reubenalfred I'm spring cleaning (e.g. https://github.com/serilog/serilog-sinks-file/pull/297#issuecomment-1770384897 https://github.com/serilog/serilog-sinks-file/pull/263#issuecomment-1770367986)

Before any work takes place to add checks, it'd be good for us to collectively get https://github.com/serilog/serilog-sinks-file/pull/258 over the line, as it may already be address some of these issues.

In short, as:
- we want/need all null checks to be complete and correct
- if there is an issue that clearly and concisely documents a desired behavior change wrt empty strings in some cases, then a PR can make the requisite changes
- in general whitespace trimming is not something that we want to do

The next steps here are to make sure your concerns are covered by either
a) commenting on #258
b) writing a new issue detailing the needs (no large screeshots please though ;))
